doc/Programming-guide/Key-platform-features/Endpoint-groups/index.md

```diff
@@ -6,166 +6,308 @@ nav: /:path/Programming-guide/Key-platform-features/Endpoint-groups
+         "otherSimpleField":789,
+         "stringField":"STRING_VALUE3"
+      }
+   ],
+   "nullableRecordField":null
+}
+```
 
+4. Let's assume the following server-side endpoint profile, which corresponds to the schema.
 
 ```json
-[
-    {
-        "namespace": "org.kaaproject.kaa.common.endpoint.gen",
-        "type": "record",
-        "name": "ServerSideEndpointProfile",
-        "fields": [
-            {
-                "name": "simpleField",
-                "type": "string"
-            },
-            {
-                "name": "arraySimpleField",
-                "type": {
-                    "type": "array",
-                    "items": "string"
-                }
-            }
-        ]
-    }
-]
+{  
+   "simpleField":"SERVER_SIDE_SIMPLE_FIELD",
+   "arraySimpleField":[  
+      "SERVER_SIDE_VALUE_1",
+      "SERVER_SIDE_VALUE_2"
+   ]
+}
 ```
 
-* Second, let's assume the following client-side endpoint profile, which corresponds to the schema.
+At last, the following filters will yield true when applied to the given endpoint.
+
+| Filter |Explanation| 
+|-------------------------------------------------------------------------------------------------------------|
+|#cp.simpleField=='CLIENT_SIDE_SIMPLE_FIELD' | The client-side endpoint profile contains simpleField with the value 'SIMPLE_FIELD'. |
+|#sp.arraySimpleField[1]=='SERVER_SIDE_VALUE_2' | The server-side endpoint profile contains arraySimpleField, which is an array containing the element 'VALUE2' in the position 
+|{'AAAAABBBBCCCDDD='}.contains(#ekh) | The endpoint key hash is AAAAABBBBCCCDDD= |
+|#cp.arraySimpleField.size()==2 | The client-side endpoint profile contains arraySimpleField, which is a collection containing two elements. |
+|#cp.recordField.otherSimpleField==123 | The client-side endpoint profile contains recordField, which is a record containing otherSimpleField set to '123'. |
+|#cp.recordField.otherMapSimpleField.size()==2 | The client-side endpoint profile contains recordField, which is a record containing otherMapSimpleField, which is a collection containing two entries. |
+|#cp.arrayRecordField[1].otherSimpleField==789 |The client-side endpoint profile contains arrayRecordField, which is an array. This array contains an element in the position 1, which is a record containing otherSimpleField set to '789'.|
+|#cp.nullableRecordField==null |An example of how to check a field for the null value.|
+|#cp.arraySimpleField[0]=='CLIENT_SIDE_VALUE_1' and # sp.arraySimpleField[0]=='SERVER_SIDE_VALUE_1'|An example of how to combine several conditions in a query.|
+|!#arrayRecordField.?[otherSimpleField==456].isEmpty() |The arrayRecordField field is an array of records. It contains at least one element that contains otherSimpleField with the value 
+
+## Using endpoint groups ##
+
+Each Kaa application has a special, built-in, non-user-editable group "all" with weight 0. The associated profile filter is automatically set equal to "true" for each profile 
```

Comment:
Could here be added some information of what other properties groups have, what is group weight, etc. Should this information be available below, please add a short description here and a reference to more details there.